### PR TITLE
New release 0.4.1

### PR DESCRIPTION
**Pull request by willmcgugan**
Replace POS websocket-library with ws4py

### CODE REVIEW

File: dataplicity/m2m/wsclient.py [hunk 1]
```diff
@@ -154,29 +154,71 @@ def get_file(self):
         return ChannelFile(self.client, self.number)
 
 
-class ThreadedDispatcher(threading.Thread, Dispatcher):
-    """Dispatches packets from a thread."""
+class WSApp(WebSocketClient):
```
Comment (willmcgugan) [inline dataplicity/m2m/wsclient.py]: An app object to avoid multiple inheritance.

File: dataplicity/m2m/wsclient.py [hunk 2]
```diff
@@ -274,10 +315,10 @@ def hard_close_channels(self):
     def run(self):
         sockopt = [(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)]
         try:
-            self.app.run_forever(sockopt=sockopt,
-                                 ping_interval=30,
-                                 ping_timeout=10,
-                                 sslopt={"cert_reqs": ssl.CERT_NONE})
+            self.app.run_forever(
```
Comment (willmcgugan) [inline dataplicity/m2m/wsclient.py]: ws4py sets `TCP_NODELAY` with by default.

File: dataplicity/m2mmanager.py [hunk 3]
```diff
@@ -111,7 +118,8 @@ def run(self):
 
         while 1:
             # Get the identity, and tell the server about it
-            identity = self._identity = self.m2m_client.wait_ready(10)
+            identity = self._identity = self.m2m_client.wait_ready(20)
```
Comment (willmcgugan) [inline dataplicity/m2mmanager.py]: bumping up timeouts a bit, to be a little kinder to the server.

**Comment by willmcgugan** [comment]
@wildfoundry/web-developers There is a tiny bit of work left on this, but please review.

**Comment by willmcgugan** [comment]
@wildfoundry/web-developers any chance you can look at this today? @toudi you are welcome to review PRs

**Comment by barfle** [comment]
@beardygeek could you please take a look?

**Comment by willmcgugan** [comment]
@beardygeek @toudi @meshy bump

**Comment by beardygeek** [comment]
👍

**Pull request merged**